Dz4

devAlexanderKonopelko opened the pull request:
Flowers app aded, access on button no.4. Used fragments, ViewModel and ect.

devAlexanderKonopelko committed:
Flowers app

Flowers app added. Worked with fragments, ViewModel and ect.

devAlexanderKonopelko committed:
Create app.iml

TMSlebedev commented on FirstApp/app/src/main/java/com/example/firstapp/flowers/Flower.kt:
```diff
@@ -0,0 +1,4 @@
+package com.example.firstapp.flowers
+
+data class Flower(val url: String?, val name: String?, val price: Double?) {
```
А почему поля nullable?

TMSlebedev commented on FirstApp/app/src/main/java/com/example/firstapp/flowers/FragmentTop.kt:
```diff
@@ -0,0 +1,43 @@
+import kotlinx.android.synthetic.main.fragment_top_layout.*
+
+class FragmentTop: Fragment() {
+    private var flower: Flower? = null
```
это что-то странное... Суть в том, что фрагмент может попросить данные из коллекции синглтона.

TMSlebedev commented on FirstApp/app/src/main/java/com/example/firstapp/flowers/MyViewModel.kt:
```diff
@@ -0,0 +1,15 @@
+class MyViewModel : ViewModel() {
+
+    fun setFlowerTopFragment(fragmentTop: FragmentTop) {
+        fragmentTop.setFlower(FlowersList.flowers.flowersList[0])
```
ViewModel точно не должна этим заниматься. Тем более очень плохая практика дергать методы у фрагмента. Я показывал, что если это требуется, то следует это делать через реализацию интерфейса фрагмента

TMSlebedev reviewed (commented):
Убрать лишнее и переделать

devAlexanderKonopelko committed:
Fragments usages fixes

Fixed issue when ViewModel used fragment methods.
Fixed issue with bullable fields in Flower class.
ViewModel deleted.
Fragments usages were simpified using singleton class.
